pkg/terraform/provider_installer_test.go

```diff
@@ -204,3 +204,31 @@ func TestProviderInstallerWithConfigDirectory(t *testing.T) {
 	assert.Equal(path.Join(fakeTmpHome, expectedSubFolder, config.GetBinaryName()), providerPath)
 
 }
+
+func TestProviderInstallerWithConfigDirectoryCantWrite(t *testing.T) {
+
+	assert := assert.New(t)
+	fakeTmpHome := t.TempDir()
+
+	expectedSubFolder := fmt.Sprintf("/.driftctl/plugins/%s_%s", runtime.GOOS, runtime.GOARCH)
+
+	config := ProviderConfig{
+		Key:       "aws",
+		Version:   "3.19.0",
+		ConfigDir: fakeTmpHome,
+	}
+
+	mockDownloader := mocks.ProviderDownloaderInterface{}
+	mockDownloader.On("Download", config.GetDownloadUrl(), path.Join(fakeTmpHome, expectedSubFolder)).Return(nil)
+
+	installer, _ := NewProviderInstaller(config)
+	installer.downloader = &mockDownloader
+
+	joinedPath := installer.getProviderDirectory()
+	_ = os.MkdirAll(path.Join(joinedPath, ".is_directory_writable"), 0000)
+
+	providerPath, err := installer.Install()
+
+	assert.Equal("", providerPath)
+	assert.EqualErrorf(err, fmt.Sprintf("can't write in configuration directory: open %s/.is_directory_writable: is a directory", joinedPath), "")
```

Comment:
`open %s/.is_directory_writable: is a directory` this error is not a permission error but just aboutt `.is_directory_writable` being a directory 

I don't get the point of this test